src/PointwiseFunctions/AnalyticSolutions/NewtonianEuler/IsentropicVortex.cpp

```diff
@@ -38,6 +39,14 @@ IsentropicVortex<Dim>::IsentropicVortex(
       equation_of_state_(1.0, adiabatic_index),
       source_term_(adiabatic_index, perturbation_amplitude, center,
                    mean_velocity, strength) {
+  if (Dim == 2) {
+    ASSERT(
+        abs(perturbation_amplitude_) < std::numeric_limits<double>::epsilon(),
```

Comment:
You can enforce this when constructed from an input file by only including `PerturbAmplitude` in the `options` list in 3D.  You'll have to overload the constructor or reorder the arguments and default `perturbation_amplitude`.  (If that doesn't work for some reason this should be a `PARSE_ERROR` because it can be triggered by an input file.)

The assertion is still good to have in case someone creates the class in C++.